src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js

```diff
@@ -40,12 +93,18 @@ define([
                 }
             }, 'toggle-cell-input-display', 'hide_input')
         ])).find('.btn').attr('id', 'btn-hide-input');
-        // Collapse all cells that are marked as hidden
-        if (Jupyter.notebook !== undefined && Jupyter.notebook._fully_loaded) {
-            // notebook already loaded. Update directly
-            update_input_visibility();
-        }
-        events.on("notebook_loaded.Notebook", update_input_visibility);
```

Comment:
This is still needed. Consider the case in which the notebook finishes loading *before* the config loads, but `Jupyter.notebook.metadata.hide_cellprompt !== undefined`. In this case, the input visibility is updated immediately, but won;t be updated again one the notebook finishes loading...

Reply:
Agree